Engine/src/Console/CVarUtil.bf

```diff
@@ -0,0 +1,182 @@
+using System;
+
+namespace SteelEngine.Console
+{
+	static class CVarUtil
+	{
+		public static bool TryParse(CVar cvar, Span<StringView> args, ref bool result, out bool didChange)
+		{
+			let currentVal = result;
+			switch (args[0])
+			{
+			case "true", "True", "TRUE", "1":
+				result = true;
+			case "false", "False", "FALSE", "0":
+				result = false;
+			default:
+				return false;
+			}
+			didChange = result != currentVal;
+			return true;
+		}
+
+		public static bool TryParse(CVar cvar, Span<StringView> args, ref int32 result, out bool didChange)
+		{
+			let currentVal = result;
+			if (int32.Parse(args[0]) case .Ok(let val))
+			{
+				result = val;
+				didChange = result != currentVal; 
+				return true;
+			}
+
+			didChange = false;
+			return false;
+		}
+
+		public static bool TryParse(CVar cvar, Span<StringView> args, ref int64 result, out bool didChange)
+		{
+			let currentVal = result;
+			if (int64.Parse(args[0]) case .Ok(let val))
+			{
+				result = val;
+				didChange = result != currentVal;
+				return true;
+			}
+
+			didChange = false;
+			return false;
+		}
+
+		public static bool TryParse(CVar cvar, Span<StringView> args, ref float result, out bool didChange)
+		{
+			let currentVal = result;
+			if (float.Parse(args[0]) case .Ok(let val))
+			{
+				result = val;
+				didChange = result != currentVal;
+				return true;
+			}
+
+			didChange = false;
+			return false;
+		}
+
+		public static bool TryParse(CVar cvar, Span<StringView> args, ref String result, out bool didChange)
+		{
+			didChange = result != args[0];
+			result.Clear();
+			result.Append(args[0]);
+			return true;
+		}
+
+		public static bool TryParse<TEnum>(CVar cvar, Span<StringView> args, ref TEnum result, out bool didChange) where TEnum : Enum
+		{
+			let currentVal = result;
+
+			if (TEnum.Parse<TEnum>(args[0], true) case .Ok(let val))
+			{
+				result = val;
+				didChange = result != currentVal;
+				return true;
+			}
+			if ((int64.Parse(args[0]) case .Ok(var iVal)) && (cvar.HasFlags(.Flags) || EnumUtils<TEnum>.HasValue(iVal)))
+			{
+				result = *(TEnum*)&iVal;
+				didChange = result != currentVal;
+				return true;
+			}
+
+			didChange = false;
+			return false;
+		}
+
+		public static int32 GetValueInt32(bool* val)
```

Comment:
All of these `GetValue*` functions assume the pointers are not null. Are we okay setting the precedent that we just assume all these values are non-null?